ecommerce/api.py

```diff
@@ -870,7 +870,11 @@ def fetch_and_serialize_unused_coupons(user):
 
     unused_coupons = []
     for coupon_data in coupons_data:
-        product = Product.objects.get(id=coupon_data["product__id"])
+        try:
+            product = Product.objects.get(id=coupon_data["product__id"])
+        except Product.DoesNotExist:
+            pass
+
```

Comment:
I checked out this code locally and it was still throwing an error because we are `passing` here instead of `continue`. 

I would suggest excluding inactive products in the above query by adding `.exclude(product__is_active=False)` at https://github.com/mitodl/mitxpro/blob/03d9aa3c30f25c29601804bbd520f4471b247986/ecommerce/api.py#L862, with this, we won't need to except exception here.  Also, a test to this change would be ideal.

Reply:
good catch, forgot to add `else` part along with the `exception`

Reply:
- Ok @tasawernawaz, At the moment, I prefer, whenever we hit the model object via `get` call then better to handle the object `DoesNotExist` too. let me update the PR.